modules/seqexec/server/src/main/scala/seqexec/server/gmos/GmosEpics.scala

```diff
@@ -3,310 +3,347 @@
 
 package seqexec.server.gmos
 
-import java.lang.{Double => JDouble}
-
+import cats.implicits._
+import cats.effect.IO
+import cats.effect.Async
+import cats.effect.Sync
 import edu.gemini.epics.acm._
-import seqexec.server.EpicsCommand.setParameter
-import seqexec.server.gmos.GmosEpics.{RoiParameters, RoiStatus}
-import seqexec.server.{EpicsCommand, EpicsSystem, ObserveCommand, SeqAction}
+import java.lang.{Double => JDouble}
+import mouse.all._
 import org.log4s.{Logger, getLogger}
-
+import seqexec.server.EpicsCommand.setParameterF
+import seqexec.server.gmos.GmosEpics.{RoiParameters, RoiStatus}
+import seqexec.server.{EpicsCommand, EpicsSystem}
+import seqexec.server.EpicsUtil._
+import seqexec.server.SeqexecFailure._
+import seqexec.server.EpicsCommandF
+import seqexec.server.ObserveCommandF
 import scala.collection.breakOut
 import scala.concurrent.duration._
-import cats.implicits._
-import mouse.all._
 
-class GmosEpics(epicsService: CaService, tops: Map[String, String]) {
+class GmosEpics[F[_]: Async](epicsService: CaService, tops: Map[String, String]) {
```

Comment:
Stylistically this would be better as a trait with unconstrained `F` and then a factory method that demands `Async`, but I realize you're trying to minimize the diff so this is fine for now.

Reply:
yeah. I may do this when we refactor `EpicsSystem` which is in a future update